view/SSViewer.php

```diff
@@ -512,32 +521,25 @@ public function __call($name, $arguments) {
 		if (isset($arguments[1]) && $arguments[1] != null) $params = $arguments[1]; 
 		else $params = array();
 
-		$hasInjected = $res = null;
-
-		if ($name == 'hasValue') {
-			if ($val = $this->getInjectedValue($property, $params, false)) {
-				$hasInjected = true; $res = (bool)$val['value'];
-			}
-		}
-		else { // XML_val
-			if ($val = $this->getInjectedValue($property, $params)) {
```

Comment:
The problem is that this line of code didn't properly handle the value already being cast to a DBField, and just did a `!= null` check on it, which of course is always true.